src/app/modules/core/components/date-time-picker/date-time-picker.component.ts

```diff
@@ -0,0 +1,178 @@
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DateTimePickerComponent),
+      multi: true
+    }
+  ]
+})
+export class DateTimePickerComponent implements ControlValueAccessor, OnInit, AfterViewInit {
+  @Input()
+  dateString: string;
+
+  @Input()
+  inputDatetimeFormat = 'M/d/yyyy H:mm:ss';
+  @Input()
+  hourStep = 1;
+  @Input()
+  minuteStep = 15;
+  @Input()
+  secondStep = 30;
+  @Input()
+  seconds = true;
+
+  @Input()
+  disabled = false;
+
+  private showTimePickerToggle = false;
+
+  private datetime: DateTimeModel = new DateTimeModel();
+  private firstTimeAssign = true;
+
+  @ViewChild(NgbDatepicker)
+  private dp: NgbDatepicker;
+
+  @ViewChild(NgbPopover)
+  private popover: NgbPopover;
+
+  private onTouched: () => void = noop;
+  private onChange: (_: any) => void = noop;
+
+  public nwisdate: string;
+
+  private ngControl: NgControl;
+  private StudyService: StudyService;
+
+  constructor(private config: NgbPopoverConfig, private inj: Injector, private studyservice: StudyService) {
+    config.autoClose = 'outside';
+    config.placement = 'left-top';
+
+    this.StudyService = studyservice;
+  }
+
+  ngOnInit(): void {
+    this.ngControl = this.inj.get(NgControl);
+  }
+
+  ngAfterViewInit(): void {
+    this.popover.hidden.subscribe($event => {
+      this.showTimePickerToggle = false;
+    });
+  }
+
+  writeValue(newModel: string) {
+    if (newModel) {
+      this.datetime = Object.assign(this.datetime, DateTimeModel.fromLocalString(newModel));
+      this.dateString = newModel;
+      this.setDateStringModel();
+    } else {
+      this.datetime = new DateTimeModel();
+    }
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  toggleDateTimeState($event) {
+    this.showTimePickerToggle = !this.showTimePickerToggle;
+    $event.stopPropagation();
+  }
+
+  setDisabledState?(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  onInputChange($event: any) {
+    const value = $event.target.value;
+    const dt = DateTimeModel.fromLocalString(value);
+
+    if (dt) {
+      this.datetime = dt;
+      this.setDateStringModel();
+    } else if (value.trim() === '') {
+      this.datetime = new DateTimeModel();
+      this.dateString = '';
+      this.onChange(this.dateString);
+    } else {
+      this.onChange(value);
+    }
+  }
+
+  onDateChange($event: any | NgbDateStruct) {
+    if ($event.year) {
+      $event = `${$event.year}-${$event.month}-${$event.day}`
+    }
+
+    const date = DateTimeModel.fromLocalString($event);
+
+    if (!date) {
+      this.dateString = this.dateString;
+      return;
+    }
+
+    if (!this.datetime) {
+      this.datetime = date;
+    }
+
+    this.datetime.year = date.year;
+    this.datetime.month = date.month;
+    this.datetime.day = date.day;
+
+    this.dp.navigateTo({ year: this.datetime.year, month: this.datetime.month });
```

Comment:
Similar to line 107, if you add something like *`if (this.dp) {this.dp.navigateTo({ year: this.datetime.year, month: this.datetime.month });}` this should do it. You won't see error about undefined navigateTo.

Reply:
This works perfectly. Thanks Tim!